src/java/org/apache/cassandra/io/util/CompressedChunkReader.java

```diff
@@ -85,28 +88,28 @@ public Rebufferer instantiateRebufferer()
 
     public static class Standard extends CompressedChunkReader
     {
-        // we read the raw compressed bytes into this buffer, then uncompressed them into the provided one.
-        private final ThreadLocal<ByteBuffer> compressedHolder;
+        // We read the raw compressed bytes into a buffer, then uncompressed them into the provided one.
+        // Notice we have 1 SimpleCachedBufferPool per BBType which wraps a FastThreadLocal BB
+        private static final EnumMap<BufferType, SimpleCachedBufferPool> reusableCompressBBs = new EnumMap<>(BufferType.class);
```

Comment:
This does not fix having to recreate buffers if the type changes, because the underlying thread local map is still a single static.
I would use the static `EnumMap` in `SimpleCachedBufferPool` instead.

Reply:
:facepalm: apologies...